cfme/cloud/instance/__init__.py

```diff
@@ -192,318 +224,558 @@ def find_quadicon(self, *args, **kwargs):
-        'Provider': provider.name
-    })
-    if template:
-        sel.click(template)
-        # In order to mitigate the sometimes very long spinner timeout, raise the timeout
-        with sel.ajax_timeout(90):
-            sel.click(form_buttons.FormButton("Continue", force_click=True))
-
-    else:
-        raise TemplateNotFound('Unable to find template "{}" for provider "{}"'.format(
-            template_name, provider.key))
+class AllView(CloudInstanceView):
+    @property
+    def is_displayed(self):
+        return (
+            self.in_cloud_instance and
+            self.title.text == 'All Instances'
+        )
 
 
 @navigator.register(Instance, 'All')
-class InstanceAll(CFMENavigateStep):
+class All(CFMENavigateStep):
+    VIEW = AllView
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
-    def am_i_here(self):
-        return match_page(summary='All Instances')
-
     def step(self):
         self.prerequisite_view.navigation.select('Compute', 'Clouds', 'Instances')
+        self.view.instances.tree.click_path('All Instances')
 
-        # use accordion
+    def resetter(self, *args, **kwargs):
         # If a filter was applied, it will persist through navigation and needs to be cleared
         if sel.is_displayed(search_box.clear_advanced_search):
             logger.debug('Clearing advanced search filter')
             sel.click(search_box.clear_advanced_search)
-        accordion.tree('Instances', 'All Instances')
+
+
+class ProviderAllView(CloudInstanceView):
+    @property
+    def is_displayed(self):
+        return (
+            self.in_cloud_instance and
+            self.title.text == 'Instances under Provider "{}"'
+                               .format(self.context['object'].provider.name)
+        )
 
 
 @navigator.register(Instance, 'AllForProvider')
-class InstanceAllForProvider(CFMENavigateStep):
+class AllForProvider(CFMENavigateStep):
+    VIEW = ProviderAllView
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
-    def am_i_here(self):
-        return match_page(summary='Instances under Provider "{}"'.format(self.obj.provider.name))
-
     def step(self, *args, **kwargs):
         self.prerequisite_view.navigation.select('Compute', 'Clouds', 'Instances')
+        self.view.instances_by_provider.tree.click_path('Instances by Provider',
+                                                        self.obj.provider.name)
 
-        # use accordion
+    def resetter(self, *args, **kwargs):
         # If a filter was applied, it will persist through navigation and needs to be cleared
         if sel.is_displayed(search_box.clear_advanced_search):
             logger.debug('Clearing advanced search filter')
             sel.click(search_box.clear_advanced_search)
-        accordion.tree('Instances by Provider', 'Instances by Provider', self.obj.provider.name)
+
+
+class AddFloatIPView(CloudInstanceView):
+    @View.nested
+    class form(ManageIQForm):
```

Comment:
@izapolsk I had separate views and forms before, but it was much too cluttered with far too many module level classes.

I've collapsed the forms into nested views, what do you think of this structure?  Note `ManageIQForm` is new, and if the team likes it I can split into a separate PR.